examples/nek/bps/bps.h

```diff
@@ -14,6 +14,10 @@
 // software, applications, hardware, advanced system engineering and early
 // testbed platforms, in support of the nation's exascale computing imperative.
 
+#ifndef bps_h
+#define bps_h
+#include <ceed.h>
```

Comment:
Have you tested this inclusion on the GPU? I feel like this could cause NVRTCC some problems.

Reply:
I tested it on OLCF Summit just now. You are right. I didn't know it could be problematic for NVRTCC. I will remove these inclusions for the other files too.

Reply:
You can likely tune your IDE to use `-include ceed.h`, which means to process as if that header was included. IMO, it would be worthwhile to extend the RTC backends (OCCA and cuda-gen) to be able to include headers instead of hand-coding some declarations into their environment.

Reply:
The problem, currently, is that including ceed.h overrides the macros that the backends themselves overrode.

Reply:
We would include a different header `ceed/qfunction.h` or some such, and it would use
```c
#ifndef MACRO
#  define MACRO default def
#endif
```

Reply:
Ah, that should be easy.

Mildly related, there is actually no technical reason for the Qfunctions to be in a .h. The CUDA backendddoes not care about the file extension. I've thought about putting the Qfunctions in .c files for the solids code so that I don't have to `make - b` or similar nonsense.

Reply:
The issue as I see it is that there may be stuff that the JIT (OCCA?) can't handle in our `.c` files.

Note that solids has dependencies so you don't need `make -B`. Try touching a header and re-running `make`.

Reply:
I was going to just rename `linElas.h` -> `linElas.c`. Identical contents.

Ah, right. That fixed that. Solids is the first code where I *don't* have to force compilation after updating the qfunction files.

Reply:
And #include it? That won't make any difference.

Just add the deps include like I did for solids.

Reply:
I was going to make a separate .h file with the definitions and include that in the main source and qfunction source. It would just make our examples look a little closer to what people expect from header files. It hasn't been a super high priority of mine though.

Reply:
In that case, the macro won't be able to provide the source file name. I'd rather have a single .c per model that initializes everything specific to that model.

Reply:
Right. Duh. I really need to catch up on some sleep and start thinking clearly.